pkg/util/system/common_linux.go

```diff
@@ -52,14 +52,16 @@ type portAndPid struct {
 
 func TCPSocks(fn netstat.AcceptFn) ([]netstat.SockTabEntry, error) {
 	socks, err := netstat.TCPSocks(fn)
-	if err != nil {
-		return nil, err
+	if err == nil && socks != nil {
+		return socks, nil
 	}
-	v6Socks, err := netstat.TCP6Socks(fn)
-	if err != nil {
-		return nil, err
+
+	socks, err = netstat.TCP6Socks(fn)
+	if err == nil && socks != nil {
+		return socks, nil
 	}
-	return append(socks, v6Socks...), nil
+
+	return nil, err
```

Comment:
Nice!  But both TCPSocks and TCP6Socks may return errors. It is recommended to return the aggregate  errors.

Reply:
> Nice! But both TCPSocks and TCP6Socks may return errors. It is recommended to return the aggregate errors.

thanks for reviewing.
done.